lib/block-supports/layout.php

```diff
@@ -582,6 +582,24 @@ function gutenberg_incremental_id_per_prefix( $prefix = '' ) {
 	return $prefix . (string) ++$id_counters[ $prefix ];
 }
 
+/**
+ * Generates a unique ID based on the structure and values of a given array.
+ *
+ * This function serializes the array into a JSON string and generates a hash
+ * that serves as a unique identifier. Optionally, a prefix can be added to
+ * the generated ID for context or categorization.
+ *
+ * @param array  $data   The input array to generate an ID from.
+ * @param string $prefix Optional. A prefix to prepend to the generated ID. Default ''.
+ *
+ * @return string The generated unique ID for the array.
+ */
+function gutenberg_unique_prefixed_id_from_array( array $data, string $prefix = '' ): string {
+	$serialized = wp_json_encode( $data );
+	$hash       = substr( md5( $serialized ), 0, 8 );
```

Comment:
Have you considered using `wp_hash` instead?

```php
$hash = substr( wp_hash( serialize( $data ) ), 0, 8 );
```

It's how it's used in WordPress core in a few places.

Reply:
Whatever is more "correct" is fine with me, haha. If the standard is to use the second one, I don't mind replacing `md5` with `wp_hash`.

We could also replace `wp_json_encode` with `serialize` (it seems you're also suggesting that, right?). I remember reading that `json_encode` was lighter than `serialize`, so I chose that function. However, I don't have a strong opinion.

Reply:
I'm not entirely sure. I copied a similar usage from the existing codebase. I just shared that to help you make the final decision. The more important aspect might be the need to pass also keys in the array with data in case we would have to change the way class names get generated in other places in the codebase to minimize the risk the same has is returned for a different set of params.

Reply:
Using `wp_hash()` is overkill because it is concerned with cryptographic security, and it combines the hashed value with a secret. If you just want a hash, then `md5()` I think is better because it is faster.

Reply:
Also, `md5()` will be consistent across all WP installs for a given input which may be important for snapshot testing. If using `wp_hash()` you'd have to make sure the underlying secrets are the same across all testing environments.

Reply:
I would check existing examples in core to see which is used most for similar hash patterns. I vaguely remember serialize being faster than json_encode, that might improved in recent PHP versions.

Reply:
PHPCS complains if you try using `serialize()` since `unserialize()` can have security problems, although this is only relevant when calling `unserialize()` with untrusted data. WPCS for PHPCS will warn when using `serialize()`:

https://github.com/WordPress/WordPress-Coding-Standards/blob/b8d2089c8c5f372fa72e0d9a726eba72c31ae7db/WordPress/Sniffs/PHP/DiscouragedPHPFunctionsSniff.php#L38-L45

So I think it's better to use `wp_json_encode()`.

Reply:
The small size of the data being serialized here most likely makes any performance difference between `serialize()` and `json_encode()` moot.

Reply:
Excellent feedback @westonruter.